### PR TITLE
[Enhancement] Blast Mask with Powder Keg

### PR DESCRIPTION
**Pull request by Caladius**
Adds an enhancement that allows the Blast Mask to utilize Powder Keg explosive properties.

### Build Artifacts
  - [2ship-linux.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/1559596890.zip)
  - [2ship-mac.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/1559596965.zip)
  - [2ship-windows.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/1559598584.zip)

### CODE REVIEW

**Comment by smashbro596** [comment]
ahh it's beautiful! thanks!

File: mm/2s2h/Enhancements/Masks/BlastMaskKeg.h [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#ifndef EQUIPMENT_BLAST_MASK_KEG_H
+#define EQUIPMENT_BLAST_MASK_KEG_H
+
+void RegisterBlastMaskKeg();
+
+#endif // EQUIPMENT_BLAST_MASK_KEG_H
```
Comment (garrettjoecox) [inline mm/2s2h/Enhancements/Masks/BlastMaskKeg.h]:
```suggestion
#endif // EQUIPMENT_BLAST_MASK_KEG_H

```

File: mm/2s2h/Enhancements/Masks/BlastMaskKeg.cpp [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+                        ? EQUIP_SLOT_B
+                        : i;
+
+                item = Player_GetItemOnButton(gPlayState, player, i);
+                if (item == ITEM_F0) {
+                    actor->shape.rot.x = BOMB_EXPLOSIVE_TYPE_POWDER_KEG;
+                }
+            }
+        });
+}
```
Comment (garrettjoecox) [inline mm/2s2h/Enhancements/Masks/BlastMaskKeg.cpp]:
```suggestion
}

```

File: mm/2s2h/BenGui/BenMenuBar.cpp [hunk 3]
```diff
@@ -521,6 +521,7 @@ void DrawEnhancementsMenu() {
         }
 
         if (UIWidgets::BeginMenu("Masks")) {
+            UIWidgets::CVarCheckbox("Blast Mask has Powder Keg Force", "gEnhancements.Masks.BlastMaskKeg", {});
```
Comment (garrettjoecox) [inline mm/2s2h/BenGui/BenMenuBar.cpp]:
```suggestion
            UIWidgets::CVarCheckbox("Blast Mask has Powder Keg Force", "gEnhancements.Masks.BlastMaskKeg");
```

**Review by garrettjoecox — APPROVED** [review]

**Pull request merged**